doc/sphinx-guides/source/developers/coding-style.rst

```diff
@@ -2,26 +2,38 @@
 Coding Style
 ============
 
-Like all development teams, the `Dataverse developers at IQSS <https://dataverse.org/about>`_ have their habits and styles when it comes to writing code. Let's attempt to get on the same page. :)
-
-.. contents:: |toctitle|
-	:local:
-
 Java
 ----
 
 Formatting Code
 ~~~~~~~~~~~~~~~
 
-Tabs vs. Spaces
-^^^^^^^^^^^^^^^
+NetBeans
+^^^^^^^^
+
+We highly recommend using Netbeans as a text editor. This makes it significantly easier to commit code that is well formatted and type safe. By default, NetBeans will warn you about some things, but if you want better warnings and code formatting, download this##################TODO ADD FILE######################### zip file, and import it to your nNtBeans settings by going to options, import, and selecting the file. These settings will cause NetBeans to automatically format code that you've worked on whenever you save, point out many common problems in code, and make sure that all indentation uses spaces.
+
+Alternatively, you can set the warnings yourself. To do so, go to tools-> options-> editor -> hints, and turn on the following
+    - Braces
+    - Error Fixes
+    - JDK Migration Support
+    - Probable Bugs
+    - Standard Javac Warnings
+
 
-Don't use tabs. Use spaces.
+Not NetBeans
+^^^^^^^^^^^^
 
-Format Code You Changed with Netbeans
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you are not going to use NetBeans, here are code style tips in order to be consistent with the rest of the project.
+- Use 4 spaces for indentation (not tabs)
+- Make sure your code is type safe.
+- Make sure not to use raw classes
+- Use diamond interface when applicable
+- Use javadoc and comments to help explain what your code is doing
+- Try to make your code look roughly like the following sample
 
-As you probably gathered from the :doc:`dev-environment` section, IQSS has standardized on Netbeans. It is much appreciated when you format your code (but only the code you touched!) using the out-of-the-box Netbeans configuration. If you have created an entirely new Java class, you can just click Source -> Format. If you are adjusting code in an existing class, highlight the code you changed and then click Source -> Format. Keeping the "diff" in your pull requests small makes them easier to code review.
+For an example code that follows our styleguide, see this
+############TODO INSERT IMAGE################################
```

Comment:
Instead of an image, let's inline a file. For an example of three downloadable JSON files, see http://guides.dataverse.org/en/4.7/installation/oauth2.html#dataverse-side